### PR TITLE
File system extended

### PR DESCRIPTION
**Pull request by aefluke**
## FileSystem Extension

Applicable helpers:

- [ ] Webdriver
- [ ] Puppeteer
- [ ] Nightmare
- [ ] REST
- [ ] Appium
- [ ] Protractor
- [ ] TestCafe

- For a project I've been working with I need to download and assert the downloaded file but filenames are partially unique, so I only know a part of the file I want to see/assert. I've added support for asserting that any file exists including the given text.
- Also for this issue I needed get the filenames in current directory, I've exploited the getFileNames() function for further use. 

## Type of change

- [ ] Breaking changes
- [x] New functionality
- [ ] Bug fix
- [ ] Documentation changes/updates
- [ ] Hot fix
- [ ] Markdown files fix - not related to source code

## Checklist:

- [x] Tests have been added
- [x] Documentation has been added (Run `robo docs`)
- [x] Lint checking (Run `npm run lint`)
- [x] Local tests are passed (Run `npm test`)

### CODE REVIEW

File: lib/helper/FileSystem.js [hunk 1]
```diff
@@ -86,6 +94,15 @@ class FileSystem extends Helper {
     const content = getFileContents(this.file, encoding);
     fileEquals(this.file).negate(text, content);
   }
+
+  /**
+  *  Returns filenames without directories.
```
Comment (DavertMik) [inline lib/helper/FileSystem.js]: Codacy Issue found: [Trailing spaces not allowed. (no-trailing-spaces)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4515295)

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4515295&bid=15213118)

File: lib/helper/FileSystem.js [hunk 2]
```diff
@@ -55,6 +55,14 @@ class FileSystem extends Helper {
     assert.ok(fileExists(this.file), `File ${name} not found in ${this.dir}`);
   }
 
+  /**
+  * Checks that any file exists including given text
+  */
+  seeFileNameMatching(text) {
+    assert.ok(this.getFileNames().some(file => file.includes(text)),
```
[the remark this answers is not in the document]
Comment (aefluke) [reply lib/helper/FileSystem.js]: I see this expression more elegant within only one line.

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4515295&bid=15213118)

File: lib/helper/FileSystem.js [hunk 3]
```diff
@@ -86,6 +94,15 @@ class FileSystem extends Helper {
     const content = getFileContents(this.file, encoding);
     fileEquals(this.file).negate(text, content);
   }
+
+  /**
+  * Returns filenames without directories.
```
Comment (DavertMik) [inline lib/helper/FileSystem.js]: Codacy Issue found: [Trailing spaces not allowed. (no-trailing-spaces)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4515295)

File: lib/helper/FileSystem.js [hunk 4]
```diff
@@ -55,6 +55,14 @@ class FileSystem extends Helper {
     assert.ok(fileExists(this.file), `File ${name} not found in ${this.dir}`);
   }
 
+  /**
+  * Checks that any file exists including given text
+  */
+  seeFileNameMatching(text) {
+    assert.ok(this.getFileNames().some(file => file.includes(text)),
+      `File contains ${text} not found in ${this.dir}`);
```
Comment (PeterNgTr) [inline lib/helper/FileSystem.js]: maybe `File name which contains ${text} not found in ${this.dir}`);` to be clearer?

**Comment by PeterNgTr** [comment]
@aefluke please rebase with master, I believe your branch is not updated since. That somehow made CI failed.

**Comment by aefluke** [comment]
Do you have any idea why circleci:docker job fails?

**Comment by PeterNgTr** [comment]
Please rebase with master Branch of codeceptjs. Your .circleci config is not updated.

File: lib/helper/FileSystem.js [hunk 5]
```diff
@@ -98,6 +106,14 @@ class FileSystem extends Helper {
     const content = getFileContents(this.file, encoding);
     fileEquals(this.file).negate(text, content);
   }
+
+  /**
+  * Returns filenames without directories.
+  */
+  getFileNames() {
```
Comment (Vorobeyko) [inline lib/helper/FileSystem.js]: this method will be added to I object. 
Can we move it from class? 
Or will need rename method on grab*

Comment (aefluke) [reply lib/helper/FileSystem.js]: I'd like it to be added to I object. Getting filenames gives freedom for you to implement more scenario varieties like validating downloaded file count, formats etc. 
Also at first I thought that grab* prefix must be about getting something from web gui. I just wanted to separate those two concepts by not naming this as grabbing. So if you think low level things like getting filenames from local disk also should be named as grabbing, I can change it, What do you think?

Comment (Vorobeyko) [reply lib/helper/FileSystem.js]: @DavertMik We need your help =)

Comment (aefluke) [reply lib/helper/FileSystem.js]: @DavertMik Any word about this mister?

Comment (PeterNgTr) [reply lib/helper/FileSystem.js]: https://codecept.io/helpers/FileSystem#docsNav

I think `grab*` prefix would make it consistent across the helper methods.

Comment (aefluke) [reply lib/helper/FileSystem.js]: Done.

File: lib/helper/FileSystem.js [hunk 6]
```diff
@@ -59,6 +59,14 @@ class FileSystem extends Helper {
     assert.ok(fileExists(this.file), `File ${name} not found in ${this.dir}`);
   }
 
+  /**
+  * Checks that any file exists including given text
```
Comment (DavertMik) [inline lib/helper/FileSystem.js]:
```suggestion
  * Checks that file with a name including given text exists in the current directory
```

Comment (DavertMik) [reply lib/helper/FileSystem.js]: Please add an example with JavaScript code here. You can use the code from your test here.

File: lib/helper/FileSystem.js [hunk 7]
```diff
@@ -98,6 +106,14 @@ class FileSystem extends Helper {
     const content = getFileContents(this.file, encoding);
     fileEquals(this.file).negate(text, content);
   }
+
+  /**
+  * Returns filenames without directories.
```
Comment (DavertMik) [inline lib/helper/FileSystem.js]:
```suggestion
  * Returns file names in current directory
```

Comment (DavertMik) [reply lib/helper/FileSystem.js]: Should also include some example

**Comment by DavertMik** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/CodeceptJS/CodeceptJS/pullRequest?prid=4515295&bid=15213118)

File: lib/helper/FileSystem.js [hunk 8]
```diff
@@ -98,6 +113,21 @@ class FileSystem extends Helper {
     const content = getFileContents(this.file, encoding);
     fileEquals(this.file).negate(text, content);
   }
+
+  /**
+  * Returns file names in current directory.
+  *
```
Comment (DavertMik) [inline lib/helper/FileSystem.js]: Codacy Issue found: [Trailing spaces not allowed. (no-trailing-spaces)](https://app.codacy.com/app/DavertMik/CodeceptJS/pullRequest?prid=4515295)

**Comment by aefluke** [comment]
Hello @DavertMik,
Examples are added, doc comments are updated as suggested.

**Comment by PeterNgTr** [comment]
Thanks @aefluke !

**Pull request merged**